### PR TITLE
[Run 1229] Assert identical commit ordering between recorded and replay

### PR DESCRIPTION
**Pull request by MichaReiser**
This PR adds an assertion to `ProxyImpl::ScheduledActionCommit` verifying that the replay and recorded agree on the performed commits (and the source frames in which the commits happen).

This assertion should help us understand if the commits diverge (in case this assertion hits) or if the same commits are happening, but the commits contain different layers (old assertion keeps hitting).

[Analysis](https://linear.app/replay/issue/RUN-1229/mismatch-in-ccpicturelayerdroprecordingsourcecontentifinvalid#comment-ee7abca4)

### CODE REVIEW

File: cc/trees/proxy_impl.cc [hunk 1]
```diff
@@ -795,6 +795,9 @@ void ProxyImpl::ScheduledActionCommit() {
 
   auto* commit_state = data_for_commit_->commit_state.get();
   auto* unsafe_state = data_for_commit_->unsafe_state.get();
+  recordreplay::Assert("[Run-1229] ProxyImpl::ScheduledActionCommit %d %d",
+                       commit_state->source_frame_number,
+                       commit_state->trace_id);
```
Comment (Domiii) [inline cc/trees/proxy_impl.cc:800]: `trace_id` is `uint64_t`, so it needs a different format specifier, e.g. `%llu`.

(AFAIK, printing a 64 bit value with a 32 bit specifier is UB)

**Review by Domiii — APPROVED** [review]

File: cc/trees/proxy_impl.cc (hunk 1 above)
Comment (MichaReiser) [reply cc/trees/proxy_impl.cc:800]: Argh, I miss Rust yelling at me when I do stupid things

Comment (Domiii) [reply cc/trees/proxy_impl.cc:800]: Funnily enough, I only just learned that our driver/linker build has warnings enabled for exactly this  😄

**Comment by Domiii** [comment]
Made some micro changes and merging it for now (see https://linear.app/replay/issue/RUN-1229#comment-2e3dd9f8)

**Pull request merged**